src/create_disk.sh

```diff
@@ -8,27 +8,65 @@
 # although see also https://github.com/coreos/coreos-assembler/pull/298
 # For people building "derived"/custom FCOS-like systems, feel free to file
 # an issue and we can discuss configuration needs.
-set -euo pipefail
 
-if [ "$#" -ne 8 ]; then
-	echo 'create_disk <device> <ostree-repo> <ostree-ref> <ostree-remote> <grub-script> <os-name> <save-var-subdirs> <space separated kargs>'
-	exit 1
-fi
+usage() {
+    cat <<EOC
+${0} create a supermin virtual machinge to create a
+Fedora CoreOS style disk image from an OS Tree.
+
+Options:
+    -R|--ostree-ref: the OSTRee reference to install
+    -d|--disk: disk device to use
+    -g|--grub-script: grub script to install
+    -h|--help: show this helper
+    -k|--kargs: kernel CLI args
+    -o|--osname: the OS name to use, e.g. fedora
+    -r|--ostree-remote: the ostree remote
+    -t|--ostree-repo: location of the ostree repo
+    -v|--save-var-subdirs: "yes" to workaround selabel issue for RHCOS
+
+You probably don't want to run this script by hand. This script is
+run as part of 'coreos-assembler build'.
+EOC
+}
+
+while [ $# -gt 0 ];
+do
+    flag="${1}"; shift;
+    arg="${1}"
+    case "${flag}" in
+        -d|--disk)             disk="${arg}"; shift;;
+        -t|--ostree-repo)      ostree="${arg}"; shift;;
+        -R|--ostree-ref)       ref="${arg}"; shift;;
+        -r|--ostree-remote)    remote_name="${arg}"; shift;;
+        -g|--grub-script)      grub_script="${arg}"; shift;;
+        -o|--osname)           os_name="${arg}"; shift;;
+        -v|--save-var-subdirs) save_var_subdirs="${arg}"; shift;;
+        -h|--help)             usage; exit;;
+        -k|--kargs)
+            # any string between -k and -- will be included
+            # as a kernel argument.
```

Comment:
The kargs are passed as one argument, so we shouldn't need any of this, right?

Reply:
That or we could make it take `-k` as many times as needed (one per karg)

Reply:
I was trying to harden the code for quoting and to provide for the use-case of quoted.

Reply:
Hmm, but using `--` to terminate a sequence of non-positional args is odd. WDYT about just handling it as one arg for now (status quo), and once we move to passing `image.yaml` directly, we can just clean this up properly?

Reply:
Removed for being too clever.